scripts/census_table.sql

```diff
@@ -1,27 +1,43 @@
 ﻿-- Database: hospital_readmission_server_dev
 
-CREATE TABLE states(
-  id SERIAL PRIMARY KEY,
-  name varchar(255),
-  population integer,
-  abbrev varchar(2)
-);
+ALTER TABLE states ADD CONSTRAINT abbrev_key UNIQUE (abbrev);
 
-SELECT * FROM hospitals
-WHERE number_of_discharges is not NULL
-LIMIT 10;
-
-ALTER TABLE states
-ADD CONSTRAINT state_fk
-FOREIGN KEY (abbrev)
-REFERENCES hospitals
-(state) MATCH FULL;
-
-SELECT name FROM states;
+# create column for state ID
+ALTER TABLE hospitals ADD COLUMN state_id integer;
 
+# update hospitals
+# state_id = s.id is an alias
+UPDATE hospitals 
+SET state_id = s.id
+FROM states s
+WHERE state = s.abbrev;
 
+SELECT * FROM states
+ORDER BY name;
 
+# add a foreign key
+ALTER TABLE hospitals
+ADD CONSTRAINT state_fk
+FOREIGN KEY (state)
+REFERENCES states
+(abbrev) MATCH FULL;
+
+# add a foreign key
+ALTER TABLE hospitals
+ADD CONSTRAINT state_idfk
+FOREIGN KEY (state_id)
+REFERENCES states
+(id) MATCH FULL;
+
+EXPLAIN ANALYZE
+SELECT * FROM hospitals
+INNER JOIN states ON (hospitals.state_id = states.id)
+WHERE states.abbrev = 'WI';
 
+SELECT * FROM hospitals
+INNER JOIN states ON (hospitals.state = states.abbrev)
+WHERE states.abbrev = 'WI';
 
 
 
+VACUUM ANALYZE;
```

Comment:
Github gets angry when you don't add a newline at the end of the file. This is mostly because when you do `cat some_file.sql` your prompt gets all screwed up if there's new newline.